lib/config/flat-config-schema.js

```diff
@@ -438,11 +454,40 @@ const sourceTypeSchema = {
     }
 };
 
+/**
+ * Creates a schema that always throws an error. Useful for warning
+ * about eslintrc-style keys.
+ * @param {string} key The eslintrc key to create a schema for.
+ * @returns {ObjectPropertySchema} The schema.
+ */
+function createEslintrcSchema(key) {
```

Comment:
Might suggest a more explicit name than `createEslintrcSchema` - the current name could be confused as _adding_ support for eslintrc config keys at a glance? Perhaps `rejectEslintrcKeySchema` or similar to spell it out?

Reply:
Fair point, making it explicit that this is throwing an error makes sense.